### ISSUE TITLE
legend

### ISSUE THREAD

**Issue by paulellison** [archimatetool/archi#267]
### Steps to Reproduce the Behaviour
1.
2.
3.

**Comment by jbsarrodie** [comment]
Hi,

Well, I guess the whole point is to be able to create a legend (there aren't any for the moment).

Question: where do you think this legend should be added (Jasper Reports, HTML export...) ?

**Comment by paulellison** [comment]
I have seen similar tools where a legend button is available in the Palette, for example maybe adding it with the other non-semantic elements (Notes and Groups).  If the Legend were able to be added to the Palette, would this allow a Legend to be picked up with the "Export Image to Clipboard" command rather than if it were added to Jasper Reports or HTML Export?

Maybe the could Legend object could have a properties checkbox allowing the user the select which elements to show in the Legend?

**Comment by theunsfvo** [comment]
This would come in very handy, because a lot of the time, diagrams and views are shared with people who are not familiar with Archimate notation... then I have to build a generic legend every time...

Would be great if there is a legend button that will add a legend for all the element types used in the view, including properties, and placing it by default in the top right of the view, and making it movable to other places in the view...

**Comment by peterprib** [comment]
Would love to add legend to my diagrams which indicates different coloring on edges and vertices to highlight what is unchanged, added, modified or removed by the architecture design.  Better still a mechanism which uses connects or tag and option of view that auto applies these colors based on work package

**Comment by colddaemon1** [comment]
Yes! This would be great. A generic legend as palette option, with customisable properties to select what to show/hide and edit.

**Comment by theunsfvo** [comment]
Even an auto generated legend of each element and relationship type in your view would help a lot, like this one...
![image](https://user-images.githubusercontent.com/5095077/59342483-2f0bc280-8d0a-11e9-990f-c7af7b95217a.png)

**Comment by MattHirdler** [comment]
+1 on this. My experience is coming from BizzDesign which has the above feature. Invaluable when sharing outside of folks intimate with Archimate.

**Comment by jbsarrodie** [comment]
FWIW, someone shared a jArchi script for that purpose. Still a bit experimental but worth to try.

See https://gist.github.com/projetnumero9/21ec861797dd6f3fcc193f8221922e8e

Don't hesitate to provide feedback on the gist page.

**Comment by nileio** [comment]
@jbsarrodie but I can't get the jArchi script to work unfortunately - I am 100% favoring an out of box solution for this.
